### PR TITLE
Update .gitignore

### PR DESCRIPTION
**Pull request by dextermb**
- Update gitignore to cover majority unwanted files
- Recommend copying of gitignore for other projects

We also probably should discuss about having a [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore).

### CODE REVIEW

File: tools.md [hunk 1]
```diff
@@ -10,3 +10,9 @@ A collection of useful tools
   * [Drop all tables within active schema](https://gist.githubusercontent.com/dextermb/3811bb599286a8ee558efc10c2f57419/raw/70d5a86154f6d96872f279060fe14e065e2f3785/drop_all_tables.sql)
   * [Remove remoteless (dead) branches](https://gist.githubusercontent.com/dextermb/04c40d02a8e73fc52c90578824ca62ff/raw/201dd731d9c4d5ee6783e965d57578908fe1b467/delete_remoteless_branches.sh)
   * [Command typo recovery](https://github.com/nvbn/thefuck)
+
+# Files
```
Comment (dextermb) [inline tools.md]: Perhaps we should create a `files.md`

File: tools.md [hunk 2]
```diff
@@ -10,3 +10,9 @@ A collection of useful tools
...
+
+A collection of useful files
+
+* [.gitignore](gitignore.md)
```
Comment (DivineOmega) [inline tools.md]: `gitignore.md` does not appear to exist?

Comment (dextermb) [reply tools.md]: As it's a relative link it goes to the PR. But viewing the branch and clicking the link works

Comment (DivineOmega) [reply tools.md]: I tried this before I commented. Clicking the link from the branch here does not work. https://github.com/langleyfoxall/technologies/blob/feature/update-gitignore/tools.md

Comment (dextermb) [reply tools.md]: 🅾️ I see why haha

Comment (dextermb) [reply tools.md]: @DivineOmega Too many markdown files 😄

**Comment by dextermb** [comment]
@wheatleyjj that would be a cool idea, although I think everything should be uncommented and the user should comment the things that they want to be left in.

As mentioned in the original post, we should also look at testing and writing a guide for a [global gitignore](https://help.github.com/articles/ignoring-files/#create-a-global-gitignore). Which would save hassle in some situations too.

**Pull request merged**